### ISSUE TITLE
href should not be available as a property

### ISSUE THREAD

**Issue by pdf** [gopherjs/vecty#170]
The behaviour of `href` is hard to predict when defined as a property, and only behaves as expected when defined as an attribute, due to the way browsers handle applying the href prop to an element.

Consider the following:

```go
elem.Div(vecty.Markup(
    vecty.MarkupIf(condition, prop.Href("link")),
))
```

If `condition` is true at any time, this will result in the href being permanently applied (and never removed), because browsers will create an attribute to match the href prop, and this attribute will never be removed, since vecty will only act on the property, that it knows about.  The following works as expected:

```go
elem.Div(vecty.Markup(
    vecty.MarkupIf(condition, vecty.Attribute("href", "link")),
))
```

There may be other props that behave in this way, but this is the only one I'm aware of currently.

**Comment by slimsag** [comment]
Interesting issue!

This sort of opens thq eustions:

1. Do we remove `prop.Href` and require users to use `vecty.Attribute("href", "link")` ? I don't like this because it's less type safe / diminishes the role of the generated subpackages.
2. Do we remove `prop.Href` and add `attr.Href`? Of course, such an `attr` package would be quite small..
3. Do we just make `prop.Href` set a an attribute instead of a property? Or is that confusing as a user that it would do that under the hood?

**Comment by pdf** [comment]
Yep, these are good questions, it may be that we still allow such things to be set as props, but we need some type-awareness to be able to correctly remove/disable them.

**Comment by slimsag** [comment]
Same issue with the `disabled` property on inputs I have found.

**Comment by AnikHasibul** [comment]
What's going on?
Any update?
